packages/@o3r/core/schematics/library/rules/rules.nx.ts

```diff
@@ -0,0 +1,183 @@
+import * as path from 'node:path';
+import { readFileSync } from 'node:fs';
+
+/**
+ * generate the rules to adapt the library generated by nx cli
+ *
+ * @param options
+ * @param rootPath
+ */
+export function nxGenerateModule(options: NgGenerateModuleSchema): Rule {
+
+  /**
+   * Determine the path where NX schematic will generate the package
+   *
+   * @param tree
+   */
+  const determineNxTargetFolder = (tree: Tree) => {
+    const { workspaceLayout } = tree.readJson('/nx.json') as { workspaceLayout: { libsDir: string } };
+    return path.posix.join(workspaceLayout.libsDir, options.name);
+  };
+
+  /**
+   * Save the content of files before the process of Nx generation
+   *
+   * @param tree
+   */
+  const savePreCommandContent = (tree: Tree) => {
+    const files = [
+      '/.vscode/extensions.json',
+      '.gitignore',
+      '.prettierignore',
+      '.prettierrc',
+      'jest.preset.js',
+      'nx.json',
+      'package.json'
+    ];
+
+    return files.reduce<Record<string, Buffer | null>>((acc, file) => {
+      if (!tree.exists(file)) {
+        acc[file] = null;
+        return acc;
+      }
+      acc[file] = tree.read(file);
+      return acc;
+    }, {});
+  };
+
+  /**
+   * Restore the file saved
+   *
+   * @param mem
+   */
+  const restoreFiles = (mem: Record<string, Buffer | null>) => (tree: Tree) => {
+    Object.entries(mem)
+      .forEach(([file, content]) => {
+        if (content === null) {
+          if (tree.exists(file)) {
+            tree.delete(file);
+          }
+        } else {
+          tree.overwrite(file, content);
+        }
+      });
+    return tree;
+  };
+
+  /**
+   * Update Nx templates
+   *
+   * @param tree File tree
+   * @param context Context of the schematics
+   */
+  const updateNxTemplate: Rule = (tree, context) => {
+    const rules: Rule[] = [];
+    let targetPath = determineNxTargetFolder(tree);
+
+    if (options.path) {
+      rules.push(move(targetPath, options.path));
+      targetPath = options.path;
+    }
+
+    const o3rCorePackageJsonPath = path.resolve(__dirname, '..', '..', '..', 'package.json');
+    const o3rCorePackageJson: PackageJson & { generatorDependencies?: Record<string, string> } = JSON.parse(readFileSync(o3rCorePackageJsonPath)!.toString());
+    const otterVersion = o3rCorePackageJson.dependencies!['@o3r/schematics'];
+
+    const templateNx = apply(url('./templates/nx'), [
+      template({
+        ...options,
+        path: targetPath,
+        projectRoot: path.posix.resolve(targetPath, options.name),
+        otterVersion,
+        runner: process.env?.npm_execpath?.includes('yarn') ? 'yarn run' : 'npm run'
+      }),
+      renameTemplateFiles(),
+      move(targetPath)
+    ]);
+    rules.push(mergeWith(templateNx, MergeStrategy.Overwrite));
+
+    return chain([
+      ...rules,
+      (t, _c) => {
```

Comment:
can't we share this rule with the one for angular ?
it seems to be the same code